contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,288 @@
+gave a preliminary design of the dynamic configuration of the list of the
+default admission controls. This document hashes out the implementation details.
+
+## Goals
+
+* Admin is able to predict what initializers/webhooks will be applied to newly
+  created objects.
+
+* Admin needs to be able to ensure initializers/webhooks config will be applied within some bound
+
+* As a fallback, admin can always restart an apiserver and guarantee it sees the latest config
+
+* Do not block the entire cluster if the intializers/webhooks are not ready
+  after registration.
+
+## Specification
+
+We assume initializers could be "fail open". We need to update the extensible
+admission control
+[proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/admission_control_extension.md)
+if this is accepted.
+
+The schema is copied from 
+[#132](https://github.com/kubernetes/community/pull/132) with a few
+modifications.
+
+```golang
+type AdmissionControlConfiguration struct {
+    metav1.TypeMeta 
+
+    // validation will only allow one name in 1.7, it can be relaxed later. 
+    v1.ObjectMeta
+
+    // ResourceInitializers is a list of resources and their default initializers
+    ResourceInitializers []ResourceDefaultInitializer
+
+    ExternalAdmissionHooks []ExternalAdmissionHook
+}
+
+// Because the order of initializers matters, and each resource might need
+// differnt order, the ResourceDefaultInitializers are indexed by Resource.
+type ResourceDefaultInitializer struct {
+    // APIGroup of the resource. Because we want to use APIGroup + Resource as a
+    // merge key, we don't use a Resource struct. See muti-fields merge key
+    // [proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/multi-fields-merge-key.md).
+    APIGroup string
+
+    // Resource identifies the type of resource to be initialized that should be
+    // initialized
+    Resource string
+
+    // Initializers are the default names that will be registered to this resource
+    Initializers []Initializer
+}
+
+type Initializer struct {
+    // Name is the string that will be registered to the resource that needs
+    // initialization.
+    Name string
+
+    // FailurePolicy defines what happens if there is no initializer controller
+    // takes action. Allowed values are Ignore, or Fail. If "Ignore" is set, 
+    // apiserver removes initilizer from the initializers list of the resource
+    // if the timeout (30s) is reached; If "Fail" is set, apiserver returns 
+    // timeout error if the timeout is reached.
+    FailurePolicy FailurePolicyType
+}
+
+type FailurePolicyType string
+
+const (
+    Ignore FailurePolicyType = "Ignore"
+    // **optional** For 1.7, only "Ignore" is allowed. We can add "Fail" when
+    // the feature is more mature.
+    Fail FailurePolicyType = "Fail"
+)
+
+type ExternalAdmissionHook struct {
```

Comment:
@whitlockjc how about this version? It solves `Overlapping Criteria Combinations`, and `Negation` can be supported easily in the future.

@lavalamp i think it's not complex for a human to read.